File: src/implementation/ShittyFortune.java
package implementation;

import java.util.Scanner;

public class ShittyFortune {

    public static void main(String[] arg){
        String str;
        StringBuilder rev = new StringBuilder();

        Scanner sc = new Scanner(System.in);

        int time = sc.nextInt();

        for (int i = 0; i < time; i++) {

            str = sc.nextLine();

            int length = str.length();

            for ( int j = length - 1; j >= 0; j-- )
                rev.append(str.charAt(j));

            if (str.equals(rev.toString()))
                System.out.println("YES");
            else
                System.out.println("NO");
        }
    }

}

File: problems/hacker_earth/implementation/ShittyFortune.kt
package hacker_earth.implementation

fun main(args: Array<String>) {

    val time = readLine()?.toInt() ?: 0
    val output = StringBuilder()

    for (t in 0 until time){

        val string = readLine() ?: ""

        var len = string.length - 1

        while (len >= 0){
            output.append(string[len])
            len--
        }

        if (string == output.toString()){
            println("YES")
        } else{
            println("NO")
        }

    }


}